tests/integration/test_package_managers.py

```diff
@@ -0,0 +1,78 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import json
+import logging
+import os
+
+import pytest
+
+from . import utils
+from .conftest import TEST_DATA_PATH, TestParameters
+
+log = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    "test_params",
+    [
+        # Test scratch build with parent from external registries
+        pytest.param(
+            TestParameters(
+                repo="https://github.com/cachito-testing/cachito-gomod-with-deps.git",
+                ref="4c65d49cae6bfbada4d479b321d8c0109fa1aa97",
+                test_case="with_deps",
+                package={"path": ".", "type": "gomod"},
+                check_files=["output.json", "fetch_deps_sha256sums.json"],
+                should_fail=False,
+                expected_rc=0,
+                expected_output="All dependencies fetched successfully",
+            ),
+        ),
+    ],
+)
+def test_packages(test_params, cachi2_image, tmpdir):
+    """
+    Test fetched dependencies for package managers.
+
+    :param TestParameters test_params: Test case arguments
+    :param str tmpdir: Temp directory for pytest
+    """
+    source_folder = utils.clone_repository(
+        test_params.repo, test_params.ref, test_params.test_case, tmpdir
+    )
+
+    output_folder = os.path.join(tmpdir, test_params.test_case)
+    cmd = [
+        "fetch-deps",
+        "--source",
+        source_folder,
+        "--output",
+        output_folder,
+        "--package",
+        json.dumps(test_params.package).encode("utf-8"),
+    ]
+    (output, rc) = cachi2_image.run_cmd_on_image(cmd, tmpdir, should_fail=False)
+    assert (
+        rc == test_params.expected_rc
+    ), f"Fetching deps end with unexpected exitcode: {rc} != {test_params.expected_rc}"
+    assert test_params.expected_output in str(
+        output
+    ), f"Expected msg {test_params.expected_output} was not found in cmd output: {output}"
+
+    if "output.json" in test_params.check_files:
+        output_json = utils.load_json(os.path.join(output_folder, "output.json"))
+        expected_output_json = utils.load_json(
+            os.path.join(TEST_DATA_PATH, test_params.test_case, "output.json")
+        )
+        log.info("Compare output.json files")
+        assert output_json == expected_output_json
+
+    if "fetch_deps_sha256sums.json" in test_params.check_files:
```

Comment:
I was thinking here that this integration test might be reused for other test cases as well and these file checks might not be relevant for them.

Reply:
Maybe it could be 
```python
check_output_json: bool = True
check_deps_checksums: bool = True
```
?

It seems weird to have a `list[str]` when only 2 exact strings actually do something

Reply:
Ack changed